File: src/test/java/no/nav/syfo/consumer/ws/PersonConsumerTest.java
package no.nav.syfo.consumer.ws;

import no.nav.tjeneste.virksomhet.person.v3.binding.HentGeografiskTilknytningPersonIkkeFunnet;
import no.nav.tjeneste.virksomhet.person.v3.binding.HentGeografiskTilknytningSikkerhetsbegrensing;
import no.nav.tjeneste.virksomhet.person.v3.binding.PersonV3;
import no.nav.tjeneste.virksomhet.person.v3.informasjon.Bydel;
import no.nav.tjeneste.virksomhet.person.v3.informasjon.Diskresjonskoder;
import no.nav.tjeneste.virksomhet.person.v3.meldinger.HentGeografiskTilknytningResponse;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class PersonConsumerTest {

    @Mock
    private PersonV3 personV3;

    @InjectMocks
    private PersonConsumer personConsumer;

    @Test
    public void hentGeografiskTilknytning() throws HentGeografiskTilknytningSikkerhetsbegrensing, HentGeografiskTilknytningPersonIkkeFunnet {

        when(personV3.hentGeografiskTilknytning(any())).thenReturn(new HentGeografiskTilknytningResponse()
                .withDiskresjonskode(null)
                .withGeografiskTilknytning(new Bydel()
                        .withGeografiskTilknytning("2017")
                ));

        GeografiskTilknytning geografiskTilknytning = personConsumer.hentGeografiskTilknytning("fnr");
        assertThat(geografiskTilknytning.geografiskTilknytning).isEqualTo("2017");
        assertThat(geografiskTilknytning.diskresjonskode).isEqualTo(null);
    }

    @Test
    public void hentGeografiskTilknytningMapperOmDiskresjonskode() throws HentGeografiskTilknytningSikkerhetsbegrensing, HentGeografiskTilknytningPersonIkkeFunnet {

        when(personV3.hentGeografiskTilknytning(any())).thenReturn(new HentGeografiskTilknytningResponse()
                .withDiskresjonskode(new Diskresjonskoder().withValue("SPSF"))
                .withGeografiskTilknytning(new Bydel()
                        .withGeografiskTilknytning("2017")
                ));

        GeografiskTilknytning geografiskTilknytning = personConsumer.hentGeografiskTilknytning("fnr");
        assertThat(geografiskTilknytning.geografiskTilknytning).isEqualTo("2017");
        assertThat(geografiskTilknytning.diskresjonskode).isEqualTo("SPSF");
    }
}

File: src/test/kotlin/no/nav/syfo/consumer/ws/PersonConsumerTest.kt
package no.nav.syfo.consumer.ws

import no.nav.tjeneste.virksomhet.person.v3.binding.HentGeografiskTilknytningPersonIkkeFunnet
import no.nav.tjeneste.virksomhet.person.v3.binding.HentGeografiskTilknytningSikkerhetsbegrensing
import no.nav.tjeneste.virksomhet.person.v3.binding.PersonV3
import no.nav.tjeneste.virksomhet.person.v3.informasjon.Bydel
import no.nav.tjeneste.virksomhet.person.v3.informasjon.Diskresjonskoder
import no.nav.tjeneste.virksomhet.person.v3.meldinger.HentGeografiskTilknytningResponse
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.ArgumentMatchers.any
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito.`when`
import org.mockito.junit.jupiter.MockitoExtension

@ExtendWith(MockitoExtension::class)
class PersonConsumerTest {

    @Mock
    private val personV3: PersonV3? = null

    @InjectMocks
    private val personConsumer: PersonConsumer? = null

    @Test
    @Throws(HentGeografiskTilknytningSikkerhetsbegrensing::class, HentGeografiskTilknytningPersonIkkeFunnet::class)
    fun hentGeografiskTilknytning() {

        `when`(personV3!!.hentGeografiskTilknytning(any())).thenReturn(
            HentGeografiskTilknytningResponse()
                .withDiskresjonskode(null)
                .withGeografiskTilknytning(
                    Bydel()
                        .withGeografiskTilknytning("2017")
                )
        )

        val geografiskTilknytning = personConsumer!!.hentGeografiskTilknytning("fnr")
        assertThat(geografiskTilknytning.geografiskTilknytning).isEqualTo("2017")
        assertThat(geografiskTilknytning.diskresjonskode).isNull()
    }

    @Test
    @Throws(HentGeografiskTilknytningSikkerhetsbegrensing::class, HentGeografiskTilknytningPersonIkkeFunnet::class)
    fun hentGeografiskTilknytningMapperOmDiskresjonskode() {

        `when`(personV3!!.hentGeografiskTilknytning(any())).thenReturn(
            HentGeografiskTilknytningResponse()
                .withDiskresjonskode(Diskresjonskoder().withValue("SPSF"))
                .withGeografiskTilknytning(
                    Bydel()
                        .withGeografiskTilknytning("2017")
                )
        )

        val geografiskTilknytning = personConsumer!!.hentGeografiskTilknytning("fnr")
        assertThat(geografiskTilknytning.geografiskTilknytning).isEqualTo("2017")
        assertThat(geografiskTilknytning.diskresjonskode).isEqualTo("SPSF")
    }

    @Test
    fun taklerAtGeografiskTilknytningErNull() {
        `when`(personV3!!.hentGeografiskTilknytning(any())).thenReturn(
            HentGeografiskTilknytningResponse()
                .withDiskresjonskode(Diskresjonskoder().withValue("SPSF"))
        )

        val geografiskTilknytning = personConsumer!!.hentGeografiskTilknytning("fnr")

        assertThat(geografiskTilknytning.geografiskTilknytning).isNull()
        assertThat(geografiskTilknytning.diskresjonskode).isEqualTo("SPSF")
    }
}
